src/AbstractOperations/computed_field.jl

```diff
@@ -89,3 +89,87 @@ end
 
     @inbounds data[i′, j′, k′] = operand[i′, j′, k′]
 end
+
+# struct FusedComputedFields{G, I}
+#     fields :: Vector{ComputedField}
+#     grid :: G
+#     indices :: I
+
+#     function FusedComputedFields(fields::Vector{ComputedField})
+#         grid = first(fields).grid
+#         indices = first(fields).indices
+#         arch = architecture(first(fields))
+#         sz = size(first(fields))
+
+#         for field in fields
+#             if !(field isa FullField) || (field isa ReducedField)
+#                 throw(ArgumentError("All fields in FusedComputedFields must be FullField!"))
+#             end
+#             if field.grid !== grid
+#                 throw(ArgumentError("All fields in FusedComputedFields must have the same grid!"))
+#             end
+#             if field.indices != indices
+#                 throw(ArgumentError("All fields in FusedComputedFields must have the same indices!"))
+#             end
+#             if architecture(field) !== arch
+#                 throw(ArgumentError("All fields in FusedComputedFields must have the same architecture!"))
+#             end
+#             if size(field) != sz
+#                 throw(ArgumentError("All fields in FusedComputedFields must have the same size!"))
+#             end
+#         end
+
+#         new(fields, grid, indices)
+#     end
+# end
+
+function compute!(comps::Tuple{Vararg{ComputedField}}, time=nothing)
+    grid = first(comps).grid
+    indices = first(comps).indices
+    arch = architecture(first(comps))
+    sz = size(first(comps))
+
+    # ordinary_field_ids = []
+
+    for (i, comp) in enumerate(comps)
+        if comp.grid !== grid
+            throw(ArgumentError("All fields to compute must have the same grid!"))
+        end
+        if comp.indices != indices
+            throw(ArgumentError("All fields to compute must have the same indices!"))
+        end
+        if architecture(comp) !== arch
+            throw(ArgumentError("All fields to compute must have the same architecture!"))
+        end
+        if size(comp) != sz
+            throw(ArgumentError("All fields to compute must have the same size!"))
+        end
+        if (comp isa FullField) && !(comp isa ReducedField)
+            # push!(ordinary_field_ids, i)
+            throw(ArgumentError("All fields to compute must be FullField!"))
+        end
+    end
```

Comment:
this looks great!! Just little more compact syntax suggestion:

```suggestion
    for (i, comp) in enumerate(comps)
        comp.grid !== grid || throw(ArgumentError("All fields to compute must have the same grid!"))
        comp.indices != indices || throw(ArgumentError("All fields to compute must have the same indices!"))
        architecture(comp) !== arch || throw(ArgumentError("All fields to compute must have the same architecture!"))
        size(comp) != sz || throw(ArgumentError("All fields to compute must have the same size!"))
        (comp isa FullField) && !(comp isa ReducedField) || throw(ArgumentError("All fields to compute must be FullField!"))
           
        end
    end
```